### PR TITLE
🎉 New Source: Pexels API [low-code cdk]

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
*Developing new connector for the source: Pexels API*
*New data source will be definitely useful for image data url's*

Resolves:
[🎉 New Source: Pexels API](https://github.com/airbytehq/connector-contest/issues/250)

## How
*Developed using low-code cdk*

## Recommended reading order
1. `spec.yaml`
2. `pexels-api.yaml`
3. `schemas/*`

## 🚨 User Impact 🚨
- No breaking changes, Just addition of new source
<details><summary><strong>New Connector: Source Pexels API</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

</details>

## Tests

<details><summary><strong>Integration & Acceptance</strong></summary>
Full Test Results

image

</details>

### CODE REVIEW

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Hello @btkcodedev, Marcos from Airbyte here 👋 . We received more than 25 new contributions along the weekend. One is yours 🎉  thank so much for! Our team is limited and maybe the review process can take longer than expected. As described in the Airbyte's Hacktoberfest your contribution was submitted before November 2nd and it is eligible to win the prize. The review process will validate other requirements. I ask to you patience until someone from the team review it. 

Because I reviewed some contributions for Hacktoberfest so far I saw some common patterns you can check in advance:
* [ ] Make sure you have added connector documentation to `/docs/integrations/`
* [ ] Remove the file `catalog` from `/integration_tests`
* [ ] Edit the `sample_config.json` inside `/integration_tests`
* [ ] For the `configured_catalog` you can use only `json_schema: {}`
* [ ] Add title to all properties in the `spec.yaml`
* [ ] Make sure the `documentationUrl` in the `spec.yaml` redirect to Airbyte's future connector page, eg: connector Airtable the `documentationUrl: https://docs.airbyte.com/integrations/sources/airtable`
* [ ] Review now new line at EOF (end-of-file) for all files.

If possible send to me a DM in Slack with the tests credentials, this process will make easier to us run integration tests and publish your connector. If you only has production keys, make sure to create a bootstrap.md explaining how to get the keys.

**Comment by marcosmarxm** [comment]
Hello! I'm going to be out of the office this Friday and won't be able to review your contribution again today, I return next Monday. So far, most contributions look solid and are almost done to be approved. As said in [Chris' comment](https://airbytehq.slack.com/archives/C0450PCSDB2/p1667430678866629) all contributions made before 2-November are eligible to receive the prize and have 2 weeks to merge the contributions. But I ensure next week we're going to have your contribution merged. If you have questions about the implementation you can send them in `#hacktoberfest-2022` in Airbyte's Slack.

Sorry the inconvenience and see you again next week, thank you so much for your contribution!

**Comment by btkcodedev** [comment]
All comments resolved. 
@marcosmarxm Review Required from community.

File: airbyte-integrations/connectors/source-pexels-api/source_pexels_api/spec.yaml
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-pexels-api/source_pexels_api/spec.yaml:33]: Remove this options. This must be configured directly by the connector.

File: airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml:40]: Remove this parameters and create a IncrementalPaginator.

File: airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml:34]: I don't think this s correct. Check how to add request parameters with other connectors implementations or readint the docs: https://docs.airbyte.com/connector-development/config-based/understanding-the-yaml-file/request-options

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments, there are some points need imprvements.

File: airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml]: Done

File: airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml]: Done

**Comment by btkcodedev** [comment]
@marcosmarxm All comments resolved.
Test results

image

File: airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml:82]: Same here. why you're searching a color?

File: airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml:35]: Why not transfer this to the stream requester and overwrite the request_parameter for that stream?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some comments.

File: airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-pexels-api/source_pexels_api/pexels_api.yaml]: Done

**Comment by btkcodedev** [comment]
My config:
{
  "api_key": "563492ad6f9170000100000126081f5bb22a4748b3575c3af4540e2f",
  "query":"people",
  "orientation": "landscape",
  "size": "large",
  "color": "orange",
  "locale": "en-US",
  "photo_id": "3756777",
  "video_id": "2499611",
  "collection_id": "mgnr9pi"
}

Recent Test Result: 

image

**Comment by btkcodedev** [comment]
@marcosmarxm All comments resolved

**Comment by marcosmarxm** [comment]
@btkcodedev I'll review your PR today!

File: airbyte-integrations/connectors/source-pexels-api/source_pexels_api/spec.yaml
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-pexels-api/source_pexels_api/spec.yaml:11]: @btkcodedev I think this still not right. I don't want to retrieve only one photo and a video and a collection. I want to probably pass a collection id and retrieve all data inside of it. 
I think most valuable for this connector it is to only implement the endpoint for search or retrieve data from collections id.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
One comment, I'm having problem running tests with current configuration.

**Comment by marcosmarxm** [comment]
Because this contribution didn't receive recent updates it won't be consider valid to hacktoberfest event. You can continue the work to get this merge in your own pace. Thank so much for your contribution!

**Comment by marcosmarxm** [comment]
> 

I'll test your changes later.

**Review by marcosmarxm — APPROVED** [review]

**Comment by marcosmarxm** [comment]
well well, @RealChrisSean base branch is locked because of code freeze. I can't merge this PR but it was published before the deadline!

Thanks for the contribution @btkcodedev

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**
